Components/Motors/VESC.cs

```diff
@@ -0,0 +1,190 @@
+        /// - Input exceeds maximum speed. Capped to given maximum.
+        /// - Filter changes value. Filter's output used instead.
+        ///     (If filter is null, this does not occur)
+        /// - The motor is disabled. You must first re-enable the motor.
+        /// </summary>
+        /// <param name="Speed"> The new speed to set the motor at. From -1.0 to 1.0 </param>
+        public void SetSpeed(float Speed)
+        {
+            if (this.Filter != null && !this.Filter.IsSteadyState() && !OngoingSpeedThread)
+            {
+                this.Filter.Feed(Speed);
+                SetSpeedThreadFactory().Start();
+                OngoingSpeedThread = true;
+            }
+            else { SetSpeedDirectly(Speed); }
+            this.TargetSpeed = Speed;
+        }
+
+        /// <summary>
+        /// Sets the speed directly given an input from -1.0 to 1.0
+        /// Takes into consideration motor stop signal and max speed restriction.
+        /// </summary>
+        /// <param name="Speed"> Speed from -1.0 to 1.0 </param>
+        private void SetSpeedDirectly(float Speed)
+        {
+            if (Speed > this.MaxSpeed) { Speed = this.MaxSpeed; }
+            if (-Speed > this.MaxSpeed) { Speed = -this.MaxSpeed; }
+            if (this.Stopped) { Speed = 0; }
+            UARTBus.Write(ConstructPacket(Speed));
+        }
+
+
+        /// <summary>
+        /// Generates the packet for the motor controller:
+        /// One Start byte (value 2 for short packets and 3 for long packets)
+        /// One or two bytes specifying the packet length
+        /// The payload of the packet
+        /// Two bytes with a CRC checksum on the payload
+        /// One stop byte (value 3)
+        /// </summary>
+        /// <param name="Speed"> Speed from -1.0 to 1.0 </param>
+        private byte[] ConstructPacket(float Speed)
+        {
+            List<byte> packet = new List<byte>();
+            List<byte> payload = new List<byte>();
+            byte ShortPacket = 2;
+
+            packet.Add(ShortPacket); // Start byte
+            
+            if (CANForwardId >= 0)
+            {
+                payload.Add((byte) PacketId.FORWARD_CAN);
+                payload.Add((byte) CANForwardId);
+            }
+
+            payload.Add((byte) PacketId.SET_DUTY);
+            // Duty Cycle (100000 mysterious magic number from https://github.com/VTAstrobotics/VESC_BBB_UART/blob/master/bldc_interface.c)
+            payload.AddRange(UtilData.ToBytes((Int32)(Speed * 100000.0)));
+
+            packet.Add((byte) payload.Count); // Length of payload
+            packet.AddRange(payload); // Payload
+
+            ushort checksum = UtilData.CRC16(payload.ToArray());
+            packet.AddRange(UtilData.ToBytes(checksum)); // Checksum
+
+            packet.Add(3); // Stop byte
+
+            return packet.ToArray();
+        }
+
+        #region enums
+        private enum PacketId : byte

```

Comment:
As per the style guidelines, abbreviations like ID should remain properly capitalized, so this should be `PacketID`.